src/uct/ib/dc/base/dc_iface.h

```diff
@@ -0,0 +1,86 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2016-.  ALL RIGHTS RESERVED.
+
+* See file LICENSE for terms.
+*/
+
+#ifndef UCT_DC_IFACE_H
+#define UCT_DC_IFACE_H
+#include <uct/ib/rc/base/rc_iface.h>
+#include <uct/ib/rc/base/rc_ep.h>
+
+#include <uct/ib/rc/verbs/rc_verbs_common.h>
+
+typedef uct_ib_qpnum_t uct_dc_iface_addr_t;
+typedef struct uct_dc_ep uct_dc_ep_t;
+
+/* TODO: derive from rc config */
+typedef struct uct_dc_iface_config {
+    /* work around to do multiple inheritance:
+     * dc_verbs needs both dc_iface_config and verbs_common_iface config 
+     */
+    uct_rc_verbs_iface_config_t   super;
+    int ndcis;
+    int tx_policy;
+    int max_inline;
+} uct_dc_iface_config_t;
+
+typedef struct uct_dc_dci {
+    uct_rc_txqp_t   txqp;
+    uct_dc_ep_t     *ep;        /* points to an endpoint that currently own
+                                   the dci */
+} uct_dc_dci_t;
+
+typedef struct uct_dc_iface { 
+    uct_rc_iface_t super;
+    struct {
+        uct_dc_dci_t  *dcis;
+        uint8_t       *dcis_stack;
+        uint8_t       stack_top;
+        uint8_t       ndci;
+        ucs_arbiter_t dci_arbiter;
+    } tx;
+    struct { 
+        struct ibv_exp_dct *dct;
+    } rx;
+    struct {
+        int tx_policy;
```

Comment:
should be enum - what are possible values? if we don't use yet - lets move to next PR